### PR TITLE
EKS and fargate config

### PR DESCRIPTION
**Pull request by igalarzab**

### CODE REVIEW

**Review by kstam — APPROVED** [review]

File: modules/eks/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+output "cluster_endpoint" {
+  value = aws_eks_cluster.eks_cluster.endpoint
+}
+
+output "kubeconfig_certificate" {
+  value = aws_eks_cluster.eks_cluster.certificate_authority.0.data
```
Comment (kstam) [inline modules/eks/outputs.tf:6]: why `.0.` ?

File: modules/eks/main.tf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  enabled_cluster_log_types = [
+    "api",
+    "audit",
+    "authenticator",
+    "controllerManager",
+    "scheduler",
+  ]
+
+  vpc_config {
+    endpoint_public_access  = false
```
Comment (kstam) [inline modules/eks/main.tf:19]: This means we will need to specify an ingress? How will we route traffic to the cluster?

File: modules/eks/vars.tf [hunk 3]
```diff
@@ -0,0 +1,13 @@
+variable "cluster_name" {
+  description = "Name of the EKS cluster and all their policies"
```
Comment (kstam) [inline modules/eks/vars.tf]:
```suggestion
  description = "Name of the EKS cluster and all its policies"
```

File: modules/eks/fargate.tf [hunk 4]
```diff
@@ -0,0 +1,42 @@
+resource "aws_iam_role" "fargate" {
+  name = "${var.cluster_name}-fargate"
+
+  assume_role_policy = <<POLICY
```
Comment (kstam) [inline modules/eks/fargate.tf]: Probably a matter of preference but I find this syntax cleaner: 

```
  assume_role_policy = jsonencode({
    Statement = [{
      Action = "sts:AssumeRole"
      Effect = "Allow"
      Principal = {
        Service = "eks-fargate-pods.amazonaws.com"
      }
    }]
    Version = "2012-10-17"
  })
```

File: modules/eks/outputs.tf (hunk 1 above)
Comment (igalarzab) [reply modules/eks/outputs.tf:6]: https://www.terraform.io/docs/providers/aws/r/eks_cluster.html

File: modules/eks/main.tf (hunk 2 above)
Comment (igalarzab) [reply modules/eks/main.tf:19]: Nop, this is for the API access. This means we'll need a bastion server to access the API. It's a security measurement (not expose the kubernetes API over internet)

**Pull request merged**